src/zenml/cli/stack.py

```diff
@@ -681,3 +683,158 @@ def down_stack(force: bool = False) -> None:
+    yaml_data = {
+        "zenml_version": zenml.__version__,
+        "stack_name": stack_name,
+        "components": component_data,
+    }
+    if filename is None:
+        filename = stack_name + ".yaml"
+    write_yaml(filename, yaml_data)
+    cli_utils.declare(f"Exported stack '{stack_name}' to file '{filename}'.")
+
+
+def _import_stack_component(
+    component_type: StackComponentType, component_config: Dict[str, str]
+) -> str:
+    """import a single stack component with given type/config"""
+    component_type = StackComponentType(component_type)
+    component_name = component_config.pop("name")
+    component_flavor = component_config.pop("flavor")
+
+    # make sure component can be registered, otherwise ask for new name
+    while True:
+        # check if component already exists
+        try:
+            other_component = _get_component_as_dict(
+                component_type, component_name
+            )
+
+        # component didn't exist yet, so we create it.
+        except KeyError:
+            break
+
+        # check whether other component has exactly same config as export
+        other_is_same = True
+        for key, value in component_config.items():
+            if key not in other_component or other_component[key] != value:
+                other_is_same = False
+                break
+
+        # component already exists and is correctly configured -> done
+        if other_is_same:
+            return component_name
+
+        # component already exists but with different config -> rename
+        display_name = _component_display_name(component_type)
+        component_name = click.prompt(
+            f"A component of type '{display_name}' with the name "
+            f"'{component_name}' already exists, "
+            f"but is configured differently. "
+            f"Please choose a different name.",
+            type=str,
+        )
+
+    _register_stack_component(
+        component_type=component_type,
+        component_name=component_name,
+        component_flavor=component_flavor,
+        **component_config,
+    )
+    return component_name
+
+
+@stack.command("import")
+@click.argument("stack_name", type=str, required=True)
+@click.argument("filename", type=str, required=False)
+@click.pass_context
+def import_stack(
+    ctx: click.Context, stack_name: str, filename: Optional[str]
+) -> None:
+    """Import a stack from YAML."""
+
+    # handle 'zenml stack import file.yaml' calls
+    if stack_name.endswith(".yaml") and filename is None:
+        filename = stack_name
+        data = read_yaml(filename)
+        stack_name = data["stack_name"]  # read stack_name from export
+
+    # standard 'zenml stack import stack_name [file.yaml]' calls
+    else:
+        # if filename is not given, assume default export name "<stack_name>.yaml"
+        if filename is None:
+            filename = stack_name + ".yaml"
+        data = read_yaml(filename)
```

Comment:
```suggestion
        data = read_yaml(filename)
        cli_utils.declare(f"Using '{filename}' to import '{stack_name}' stack.")
```

Just something small like this?